File: apps/android/shared/src/main/java/com/floently/shared/billing/FloentlyAccessService.kt
package com.floently.shared.billing

import com.floently.shared.api.FloentlyApiClient

class FloentlyAccessService(
    private val api: FloentlyApiClient
) {
    suspend fun fetchStatus(): FloentlyAccessStatus {
        return accessStatusFromJson(api.get("/api/v1/subscription/status"))
    }
}

File: apps/ios/FloentlyShared/Sources/FloentlyShared/Billing/FloentlyAccessService.swift
import Foundation

public final class FloentlyAccessService {
    private let api: FloentlyAPIClient

    public init(api: FloentlyAPIClient) {
        self.api = api
    }

    public func fetchStatus() async throws -> FloentlyAccessStatus {
        try await api.get("/api/v1/subscription/status", as: FloentlyAccessStatus.self)
    }
}
